### PR TITLE
Add data root field inside header extrinsic root

### PR DESCRIPTION
**Pull request by prabal-banerjee**

### CODE REVIEW

File: pallets/system/src/header_builder.rs [hunk 1]
```diff
@@ -158,3 +180,86 @@ pub trait HostedHeaderBuilder {
 		)
 	}
 }
+
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct AvailExtrinsic {
```
Comment (aterentic-ethernal) [inline pallets/system/src/header_builder.rs:185]: Maybe it makes sense to move this struct to da-primitives and reuse it in both header builder and rpc call?

**Review by aterentic-ethernal — APPROVED** [review]
Looks good to me, just formatting and if its possible, we should reuse structs.

**Review by aterentic-ethernal — CHANGES_REQUESTED** [review]
There is also a test in runtime/src/lib.rs which fails to compile, so changes has to be made before merge

**Comment by kroos47** [comment]
> There is also a test in runtime/src/lib.rs which fails to compile, so changes has to be made before merge

yeh actually multiple other tests needs to be fixed, so working on it. 😄

**Review by kroos47 — APPROVED** [review]

**Pull request merged**